lib/product-import.coffee

```diff
@@ -230,6 +232,52 @@ class ProductImport
       else
         false
 
+  _updateProductRepeater: (prodToProcess, existingProduct) ->
+    repeater = new Repeater {attempts: 5}
+    repeater.execute =>
+      @_updateProduct(prodToProcess, existingProduct)
+    , (e) =>
+      if e.statusCode isnt 409 # concurrent modification
+        return Promise.reject e
+
+      @logger.warn "Recovering from 409 concurrentModification error on product '#{existingProduct.id}'"
+
+      Promise.resolve => # next task must be a function
+        @client.productProjections.staged(true).byId(existingProduct.id).fetch()
+        .then (result) =>
+          @_updateProduct(prodToProcess, result.body)
+
+  _updateProduct: (_prodToProcess, existingProduct) ->
+    prodToProcess = _.deepClone(_prodToProcess)
```

Comment:
Code refactored here 3f47b7c - Products are not preprocessed again and thus we don't need a deepClone anymore.